app/controllers/V1/submissions/searches_controller.rb

```diff
@@ -0,0 +1,144 @@
+module V1
+  module Submissions
+    class SearchesController < ApplicationController
+      SORTABLE_COLUMNS = %w[laa_reference firm_name client_name date_updated status_with_assignment].freeze
+
+      def create
+        @data = search_query
+
+        render json: build_results, status: :created
+      rescue StandardError => e
+        Rails.logger.fatal("AppStore search query raised #{e.message}")
+        render json: { message: "AppStore search query raised #{e.message}" }, status: :unprocessable_entity
+      end
+
+    private
+
+      def search_query
+        relation = Search.where(date_submitted: (submitted_from..submitted_to))
+        relation = relation.where(date_updated: (updated_from..updated_to))
+        relation = relation.where(application_type:) if application_type
+        relation = relation.where(status_with_assignment:) if status_with_assignment
+        relation = relation.where("has_been_assigned_to ? :caseworker_id", caseworker_id:) if caseworker_id
+        relation = relation.order(sort_clause)
+
+        relation.where_terms(query)
+      end
+
+      def build_results
+        {
+          metadata: {
+            total_results: @data.size,
+            page:,
+            per_page:,
+          },
+          data: @data.limit(limit).offset(offset),
+          raw_data: raw_data_for_page,
+        }.to_json
+      end
+
+      def raw_data_for_page
+        # Ensures sorting and paginating works for the raw_data query.
+        # Paginate (offset/limit) here just to get an array of ordered ids for
+        # the page then query the "raw data" for just those ids, sorting by the
+        # order they were in from the "data".
+        ids = @data.limit(limit).offset(offset).pluck(:id)
+
+        Submission.find(ids).sort_by do |submission|
+          ids.index(submission.id)
+        end
+      end
+
+      def application_type
+        search_params[:application_type]
+      end
+
+      def submitted_from
+        search_params[:submitted_from]&.to_date
+      end
+
+      def submitted_to
+        search_params[:submitted_to]&.to_date
+      end
+
+      def updated_from
+        search_params[:updated_from]&.to_date
+      end
+
+      def updated_to
+        search_params[:updated_to]&.to_date
+      end
+
+      def status_with_assignment
+        search_params[:status_with_assignment]
+      end
+
+      def caseworker_id
+        search_params[:caseworker_id]
+      end
+
+      def query
+        search_params[:query]
+      end
+
+      def sort_clause
+        return "date_updated desc" unless search_params[:sort_by]
+        raise "Unsortable column \"#{sort_by}\" supplied as sort_by argument" unless SORTABLE_COLUMNS.include?(sort_by.downcase)
+
+        if sort_by == "date_updated"
+          "#{sort_by} #{sort_direction}"
+        else
+          "LOWER(#{sort_by}) #{sort_direction}"
```

Comment:
so I don;t think this `lower` shouldn't be needed... 

as the most fields are stored in lower case and those that aren't (search_field) are already converted to lowercase in the virtual column definition

Reply:
This is specifically for sorting the found/returned results. I certainly find that locally a lower case name appears after or before ALL upper case names, for example. The database collation type of which ever database is in use could also be the issue but a simple "lower" avoids the problem across any database.